components/Main.js

```diff
@@ -0,0 +1,102 @@
+import { react, html, css } from 'https://unpkg.com/rplus-production@1.0.0';
+import { useStateValue } from '../utils/globalState.js';
+import { parseUrl } from '../utils/parseUrl.js';
+import { parseDependencies } from '../utils/recursiveDependencyFetch.js';
+
+import Nav from './Nav.js';
+import Article from './Article.js';
+
+const replaceState = url => history.replaceState(null, null, url);
+
+export default () => {
+  const [state, dispatch] = useStateValue();
+  const { request, code, packagesSearchTerm } = state;
+
+  // Update the request on user navigation
+  react.useEffect(() => {
+    const updateRequest = () => {
+      dispatch({ type: 'setRequest', payload: parseUrl() });
+    };
+    addEventListener('popstate', updateRequest);
```

Comment:
As best practice, can we clean up this event listener as the return of this `useEffect`?